rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -155,6 +155,28 @@ func (p *BatchProposer) TryProposeBatch() {
 }
 
 func (p *BatchProposer) updateDBBatchInfo(batch *encoding.Batch, codecVersion encoding.CodecVersion, metrics utils.BatchMetrics) error {
+	for {
```

Comment:
So we would come into this loop only when we have enough chunks in the batch, is it right?

Currently, which factors would constraint the maxium number of chunks?

Reply:
yes.

Blob size, uncompressed batch size, first block's timestamp, calldata size and commit tx's gas limit will all be treated as constraints of proposing a batch.